From d68365a35dd00cced7cdc6eff2ad8a3b823c26f6 Mon Sep 17 00:00:00 2001
From: Michael Calmer <mc@suse.de>
Date: Thu, 7 Dec 2017 17:30:31 +0100
Subject: [PATCH] older logrotate need su directive

---
 pkg/suse/salt-common.logrotate | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pkg/suse/salt-common.logrotate b/pkg/suse/salt-common.logrotate
index 0d99d1b801..625670942c 100644
--- a/pkg/suse/salt-common.logrotate
+++ b/pkg/suse/salt-common.logrotate
@@ -8,6 +8,7 @@
 }
 
 /var/log/salt/minion {
+	su root root
 	weekly
 	missingok
 	rotate 7
-- 
2.15.1


