From 7f207ed0aa45063e90db2452a311b91f34b1b7e2 Mon Sep 17 00:00:00 2001
From: Johannes Renner <jrenner@suse.com>
Date: Thu, 11 Jan 2018 15:55:25 +0100
Subject: [PATCH] Set SHELL environment variable (#64)

---
 pkg/suse/salt-api.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pkg/suse/salt-api.service b/pkg/suse/salt-api.service
index 6634b74a7d..5c73bb9022 100644
--- a/pkg/suse/salt-api.service
+++ b/pkg/suse/salt-api.service
@@ -6,6 +6,7 @@ After=network.target
 [Service]
 User=salt
 Type=simple
+Environment=SHELL=/bin/bash
 LimitNOFILE=8192
 ExecStart=/usr/bin/salt-api
 TimeoutStopSec=3
-- 
2.13.6


