From 1cd36e85f2fbc63976e1c34901320503f627ac55 Mon Sep 17 00:00:00 2001
From: Bo Maryniuk <bo@maryniuk.net>
Date: Wed, 25 Jul 2018 11:50:28 +0200
Subject: [PATCH] Add missing dateutils import (bsc#1099945) (#96)

---
 salt/utils/templates.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/salt/utils/templates.py b/salt/utils/templates.py
index 09984c9594..56cce888b1 100644
--- a/salt/utils/templates.py
+++ b/salt/utils/templates.py
@@ -28,6 +28,7 @@ else:
 
 # Import Salt libs
 import salt.utils.data
+import salt.utils.dateutils
 import salt.utils.http
 import salt.utils.files
 import salt.utils.platform
-- 
2.17.1


