From 106864092d75da2b41098b3cc0fdcb604b6f6aaa Mon Sep 17 00:00:00 2001
From: Bo Maryniuk <bo@suse.de>
Date: Mon, 19 Jun 2017 14:53:46 +0200
Subject: [PATCH] Bugfix: use UTC date

---
 salt/modules/zypper.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/salt/modules/zypper.py b/salt/modules/zypper.py
index 2299d66dbc..63e219987a 100644
--- a/salt/modules/zypper.py
+++ b/salt/modules/zypper.py
@@ -1852,7 +1852,7 @@ def list_downloaded():
                 'path': package_path,
                 'size': os.path.getsize(package_path),
                 'creation_date_time_t': pkg_timestamp,
-                'creation_date_time': datetime.datetime.fromtimestamp(pkg_timestamp).isoformat(),
+                'creation_date_time': datetime.datetime.utcfromtimestamp(pkg_timestamp).isoformat(),
             }
     return ret
 
-- 
2.13.6


