From 89e9115ca8e2471149b75d418e780cc8d7db339e Mon Sep 17 00:00:00 2001
From: "Gareth J. Greenaway" <gareth@wiked.org>
Date: Tue, 22 Jan 2019 14:04:31 -0800
Subject: [PATCH] Ensure metadata is an accepted keyword argument for
 authentication functions.

---
 salt/auth/__init__.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/salt/auth/__init__.py b/salt/auth/__init__.py
index c4e44097f3..ca7168d00e 100644
--- a/salt/auth/__init__.py
+++ b/salt/auth/__init__.py
@@ -46,6 +46,7 @@ AUTH_INTERNAL_KEYWORDS = frozenset([
     'gather_job_timeout',
     'kwarg',
     'match',
+    'metadata',
     'print_event',
     'raw',
     'yield_pub_data'
-- 
2.17.1


