From 383cc2b4dfb3b683119d82982efe5efc5f0a24eb Mon Sep 17 00:00:00 2001
From: Mihai Dinca <mdinca@suse.de>
Date: Thu, 2 May 2019 10:50:17 +0200
Subject: [PATCH] Fix syndic start issue

---
 salt/utils/event.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/salt/utils/event.py b/salt/utils/event.py
index 744bd62bf9..ccd1b3e6a4 100644
--- a/salt/utils/event.py
+++ b/salt/utils/event.py
@@ -874,7 +874,7 @@ class SaltEvent(object):
         self.subscriber.callbacks.add(event_handler)
         if not self.subscriber.reading:
             # This will handle reconnects
-            self.subscriber.read_async()
+            return self.subscriber.read_async()
 
     def __del__(self):
         # skip exceptions in destroy-- since destroy() doesn't cover interpreter
-- 
2.16.4


