From 09c3efb34fa4e4bf3bf1919d9f1e00645a4ccae1 Mon Sep 17 00:00:00 2001
From: Victor Zhestkov <35733135+vzhestkov@users.noreply.github.com>
Date: Tue, 13 Apr 2021 18:00:42 +0300
Subject: [PATCH] Regression fix of salt-ssh on processing targets
 (#353)

---
 salt/client/ssh/__init__.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/salt/client/ssh/__init__.py b/salt/client/ssh/__init__.py
index 910f6adb95..be5af46bc4 100644
--- a/salt/client/ssh/__init__.py
+++ b/salt/client/ssh/__init__.py
@@ -426,8 +426,6 @@ class SSH(object):
             self.opts["tgt"] = _hosts
         elif _hosts:
             self.opts["tgt"] = _hosts[0]
-        else:
-            self.opts["tgt"] = ""
 
     def get_pubkey(self):
         '''
-- 
2.30.2


