From f093c7e4dc50b6c476e6687c0df113018a44f683 Mon Sep 17 00:00:00 2001
From: Mihai Dinca <mdinca@suse.de>
Date: Tue, 13 Feb 2018 16:11:20 +0100
Subject: [PATCH] Remove obsolete unicode handling in pkg.info_installed

---
 salt/modules/zypper.py | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/salt/modules/zypper.py b/salt/modules/zypper.py
index b831142c83..aa18bbe558 100644
--- a/salt/modules/zypper.py
+++ b/salt/modules/zypper.py
@@ -308,7 +308,7 @@ class _Zypper(object):
         if self.error_msg and not self.__no_raise and not self.__ignore_repo_failure:
             raise CommandExecutionError('Zypper command failure: {0}'.format(self.error_msg))
 
-        return self._is_xml_mode() and dom.parseString(self.__call_result['stdout']) or self.__call_result['stdout']
+        return self._is_xml_mode() and dom.parseString(self.__call_result['stdout'].encode('utf-8')) or self.__call_result['stdout']
 
 
 __zypper__ = _Zypper()
@@ -480,16 +480,7 @@ def info_installed(*names, **kwargs):
     for pkg_name, pkg_nfo in __salt__['lowpkg.info'](*names, **kwargs).items():
         t_nfo = dict()
         # Translate dpkg-specific keys to a common structure
-        for key, value in pkg_nfo.items():
-            if isinstance(value, six.string_types):
-                # Check, if string is encoded in a proper UTF-8
-                if six.PY3:
-                    value_ = value.encode('UTF-8', 'ignore').decode('UTF-8', 'ignore')
-                else:
-                    value_ = value.decode('UTF-8', 'ignore').encode('UTF-8', 'ignore')
-                if value != value_:
-                    value = kwargs.get('errors', 'ignore') == 'ignore' and value_ or 'N/A (invalid UTF-8)'
-                    log.error('Package {0} has bad UTF-8 code in {1}: {2}'.format(pkg_name, key, value))
+        for key, value in six.iteritems(pkg_nfo):
             if key == 'source_rpm':
                 t_nfo['source'] = value
             else:
-- 
2.16.1


