From 2e7358326d91c4bb97275f82c06ad35d40cac2fa Mon Sep 17 00:00:00 2001
From: Mihai Dinca <mdinca@suse.de>
Date: Tue, 18 Dec 2018 13:39:56 +0100
Subject: [PATCH] Strip architecture from package name in aptpkg module

---
 salt/modules/aptpkg.py | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/salt/modules/aptpkg.py b/salt/modules/aptpkg.py
index c54fd87424..24a9842b85 100644
--- a/salt/modules/aptpkg.py
+++ b/salt/modules/aptpkg.py
@@ -218,6 +218,23 @@ def _warn_software_properties(repo):
     log.warning('Best guess at ppa format: %s', repo)
 
 
+def normalize_name(name):
+    '''
+    Strips the architecture from the specified package name, if necessary.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.normalize_name zsh:amd64
+    '''
+    try:
+        name, arch = name.rsplit(':', 1)
+    except ValueError:
+        return name
+    return name
+
+
 def latest_version(*names, **kwargs):
     '''
     Return the latest version of the named package available for upgrade or
-- 
2.19.1


